idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/folding/foldingFileEditorListener.kt

```diff
@@ -0,0 +1,36 @@
+package arrow.meta.ide.plugins.proofs.folding
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.codeInsight.folding.CodeFoldingManager
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.EditorFactory
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.vfs.VirtualFile
+
+val IdeMetaPlugin.foldingFileEditor: ExtensionPhase
+  get() = addFileEditorListener(
+    fileOpened = { _: FileEditorManager, _: VirtualFile, _: FileEditor, document: Document ->
+      EditorFactory.getInstance().getEditors(document).mapNotNull { editor ->
+        editor.caretModel.addCaretListener(addEditorCaretListener(
```

Comment:
abstract this part into `EditorSyntax`. So that there is a function
`IdeMetaPlugin.addCaretListener(...): ExtensinoPhase` removing the registry logic

Reply:
At the end, the algebra only provides the parameters in the current `addEditorCaretListener`, which shall be renamed to `caretListener`